### ISSUE TITLE
opam show -f nonexist isn't an error and many fields aren't supported

### ISSUE THREAD

**Issue by dsheets** [ocaml/opam#2122]
```
$ cat ../opam-repository/packages/openflow/openflow.0.9.1/opam
opam-version: "1.2"
ocaml-version: [ >= "4.02.0" ]
version: "0.9.1"
name: "openflow"
maintainer: "Arjun Guha <arjun@cs.umass.edu>"
build: [
  ["./configure" "--%{pa_ounit:enable}%-tests"]
  [make]
  [make "install"]
]
build-test: [
  [make "test"]
]
remove: [
  ["ocamlfind" "remove" "openflow"]
]
depends: [
  "ocamlfind" { build }
  "core"
  "cstruct" {>= "1.0.1"}
  "packet"  {>= "0.3.1" }
  "sexplib"
  "async" {>= "112.17.00" }
  "quickcheck"
  "ounit" {test}
  "pa_ounit" {test}
]
$ opam show -f build-testasdf openflow
$ echo $?
0
$ opam show -f build-test openflow
$ echo $?
0
$ opam show -f opam-version openflow
$ opam show -f name openflow
$ opam show -f maintainer openflow
$ opam show -f build openflow
$ opam show -f remove openflow
```

In particular, the `build-test` field's existence is interesting for test automation to determine if further rebuilds are necessary. See <https://github.com/ocaml/ocaml-travisci-skeleton#running-the-tests>.

**Comment by AltGr** [comment]
The field names correspond to the `opam show` basic output:
```
% opam show openflow.0.9.1
             package: openflow
             version: 0.9.1
          repository: local
        upstream-url: https://github.com/frenetic-lang/ocaml-openflow/archive/v0.9.1.tar.gz
       upstream-kind: http
   upstream-checksum: 16eb012526501cf28729ef873b359889
             depends: ocamlfind & core & cstruct >= 1.0.1 & packet >= 0.3.1 & sexplib & async >= 112.17.00 & quickcheck
   installed-version: 
  available-versions: 0.1, 0.1.1, 0.2.0, 0.3.0, 0.4.0, 0.5.0, 0.6.0, 0.6.1, 0.6.2, 0.7.0, 0.8.0, 0.9.0, 0.9.1
         description: Serialization and protocol implementation for OpenFlow 1.{0,3}

This library provides the basic building blocks that are necessary to build an
OpenFlow 1.0 or 1.3.4 based network controller, including serialzation
functions and async-based protocol implementations.
```
Indeed, it's quite limited, and titles should be unified -- or rather, fileds from the opam file added, since this is mostly information of a different nature (related to the url / descr files and the current installation/switch/repository).

> `$ cat ../opam-repository/packages/openflow/openflow.0.9.1/opam`

Note that you can do `opam show --raw` (or even `cat $(opam show --where)`) for the same result :)

**Comment by AltGr** [comment]
`opam show` has recently been vastly improved, with a larger selection of fields. opam-file fields can be specified (suffixed with `:`, which is the general convention), and `opam show` specific fields can be specified as well.
